### PR TITLE
Add logger config module

### PR DESCRIPTION
**Pull request by agrberg**

### CODE REVIEW

**Review by dblock — COMMENTED** [review]
This is really good! Should we move things to `Config/Client` and `Config/Logger`?

**Comment by agrberg** [comment]
That makes sense. I didn't go down a deeper path to get the work done sooner. Would they both get included into `Client` for something like

```ruby
IEX::Api::Client.logger do |logger|
  logger.instance = Logger.new(STDOUT)
  logger.options = { bodies: true }
  logger.proc = proc { |logger| logger.filter(/T?[sp]k_\w+/i, '[REMOVED]') }
end
# and for compatibility (needed any longer?)
IEX::Api::Client.logger = Logger.new(STDOUT)
```

**Comment by dblock** [comment]
Yes, I think we want all these variations.

File: lib/iex/api/client.rb [hunk 1]
```diff
@@ -22,12 +22,15 @@ class Client
...
+      include Config::Client
+
+      attr_reader :logger
 
       def initialize(options = {})
-        Config::ATTRIBUTES.each do |key|
+        Config::Client::ATTRIBUTES.each do |key|
           send("#{key}=", options[key] || IEX::Api.config.send(key))
         end
+        @logger = Config::Logger.dup
```
Comment (agrberg) [inline lib/iex/api/client.rb:33]: Not sure if this is the best approach but similar to the block above, we want to cache a local copy of the logger config this `Client` was instantiated with. Duplicating the module to enable the same dot access was the fastest way but another object could be build or a `Struct` used.

**Comment by dblock** [comment]
👏 @agrberg want to help out with co-maintaining this library? If yes email me your rubygems email to dblock[at]dblock[dot]org and I'll give you r/w permissions. Maybe you can publish the next release?

**Pull request merged**